nntrainer/layers/cl_layers/rmsnorm_layer_cl.cpp

```diff
@@ -0,0 +1,384 @@
+  } else{
+    rmsnormProcess_fp16(in, out, gamma, epsilon, context);
+  }
+}
+
+opencl::Kernel RMSNormLayerCl::kernel_rmsnorm;
+opencl::Kernel RMSNormLayerCl::kernel_rmsnorm_fp16;
+
+void RMSNormLayerCl::rmsnormProcess(Tensor const &input, Tensor &result,
+                                    Tensor const &gamma, const float epsilon,
+                                    RunLayerContext &context) {
+  bool ret = false; 
+  int dim1 = input.batch() * input.height() * input.width() * input.channel();
+  CREATE_IF_EMPTY_DIMS(result, input.batch(), input.channel(), input.height(),
+                       input.width(), input.getTensorType());
+  int b = input.batch();
+  int c = input.channel();
+  int h = input.height();
+  int w = input.width();
+  do {
+    ret =
+      context.clCreateKernel(rmsnorm_cl_kernel_, context.LayerKernel::RMSNORM,
+                             RMSNormLayerCl::kernel_rmsnorm);
+    if (!ret) {
+      break;
+    }
+
+    opencl::Buffer inputbuf(context.context_inst_, dim1 * sizeof(float), true,
+                            nullptr);
+
+    opencl::Buffer gammabuf(context.context_inst_,
+                            input.width() * sizeof(float), true, nullptr);
+    opencl::Buffer resultbuf(context.context_inst_, dim1 * sizeof(float), true,
+                             nullptr);
+
+    const float *data = input.getData();
+    float *rdata = result.getData();
+    const float *gdata = gamma.getData();
+    ret = inputbuf.WriteData(context.command_queue_inst_, data);
+    if (!ret) {
+      break;
+    }
+
+    ret = gammabuf.WriteData(context.command_queue_inst_, gdata);
+    if (!ret) {
+      break;
+    }
+    ret = RMSNormLayerCl::kernel_rmsnorm.SetKernelArguments(0, &inputbuf,
+                                                            sizeof(cl_mem));
+    if (!ret) {
+      break;
+    }
+
+    ret = RMSNormLayerCl::kernel_rmsnorm.SetKernelArguments(1, &resultbuf,
+                                                            sizeof(cl_mem));
+    if (!ret) {
+      break;
+    }
+    
+    ret = RMSNormLayerCl::kernel_rmsnorm.SetKernelArguments(2, &gammabuf,
+                                                            sizeof(cl_mem));
+    if (!ret) {
+      break;
+    }
+    ret = RMSNormLayerCl::kernel_rmsnorm.SetKernelArguments(4, &b, sizeof(int));
+
+    if (!ret) {
+      break;
+    }
+
+    ret = RMSNormLayerCl::kernel_rmsnorm.SetKernelArguments(3, &epsilon,
+                                                            sizeof(float));
+    if (!ret) {
+      break;
+    }
+
+    ret = RMSNormLayerCl::kernel_rmsnorm.SetKernelArguments(5, &c, sizeof(int));
+    if (!ret) {
+      break;
+    }
+
+    ret = RMSNormLayerCl::kernel_rmsnorm.SetKernelArguments(6, &h, sizeof(int));
+    if (!ret) {
+      break;
+    }
+    ret = RMSNormLayerCl::kernel_rmsnorm.SetKernelArguments(7, &w, sizeof(int));
+    if (!ret) {
+      break;
+    }
+    const int work_groups_count[3] = {b * h * c, 1, 1};
```

Comment:
Here's the related code of my question above.

Reply:
Hello Addressed the reviews in latest commit. please check